Patch-Source: https://gitlab.freedesktop.org/upower/upower/-/merge_requests/252
---
From 0074161eb7318ef851f34a3d36e4d53851a2ad5c Mon Sep 17 00:00:00 2001
From: fossdd <fossdd@pwned.life>
Date: Fri, 10 Jan 2025 17:42:14 +0100
Subject: [PATCH] Fix test race in test_bluetooth_hidpp_mouse

We ship the proposed patch already since Oct 2024 in Alpine Linux and is
working for us correctly and fixes the test.

Closes https://gitlab.freedesktop.org/upower/upower/-/issues/228

Co-Authored-By: Bastien Nocera <hadess@hadess.net>
---
 src/linux/integration-test.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/linux/integration-test.py b/src/linux/integration-test.py
index 020e6b7d..7ddd975b 100755
--- a/src/linux/integration-test.py
+++ b/src/linux/integration-test.py
@@ -4082,6 +4082,7 @@ class Tests(dbusmock.DBusTestCase):
         bat0_up = devs[0]
 
         # Check we have the Bluetooth name
+        self.assertEventually(lambda: self.get_dbus_dev_property(bat0_up, 'Model'), value=alias)
         self.assertEqual(self.get_dbus_dev_property(bat0_up, "Model"), alias)
         # Check we have the kernel percentage
         self.assertEqual(self.get_dbus_dev_property(bat0_up, "Percentage"), 30)
-- 
GitLab

